controllers/etcd_controller.go

```diff
@@ -262,7 +318,15 @@ func (r *EtcdReconciler) delete(ctx context.Context, etcd *druidv1alpha1.Etcd) (
 		}, err
 	}
 
-	if err := r.removeFinalizersToDependantSecrets(ctx, etcd); err != nil {
+	if etcd.Spec.VolumeRetentionPolicy == nil || *etcd.Spec.VolumeRetentionPolicy == druidv1alpha1.VolumeRetentionPolicyDelete {
+		if err := r.removePVCs(pvcList.Items); err != nil {
+			return ctrl.Result{
+				Requeue: true,
```

Comment:
Shouldn't we `RequeueAfter` to give time to whatever it is that is preventing the PVC deletion?